include/sys/vm_amap.h

```diff
@@ -1,45 +1,116 @@
 #ifndef _SYS_VM_AMAP_H_
 #define _SYS_VM_AMAP_H_
 
+#include <sys/mutex.h>
 #include <sys/types.h>
 #include <sys/vm.h>
 #include <stddef.h>
 
 typedef struct vm_amap vm_amap_t;
+typedef struct vm_anon vm_anon_t;
 typedef struct vm_aref vm_aref_t;
 
+/* Anon structure
+ *
+ * Marks for fields locks:
+ *  (@) guarded by vm-amap::mtx
+ *  (!) read-only access, do not modify!
+ */
+struct vm_anon {
+  mtx_t mtx;        /* Anon lock */
+  uint32_t ref_cnt; /* (@) number of references */
+  vm_page_t *page;  /* (!) anon page */
+};
+
 struct vm_aref {
   size_t offset;   /* offset in slots */
   vm_amap_t *amap; /* underlying amap */
 };
 
-/*
- * Allocate new amap with specified number of slots.
+/** Allocate new amap.
  *
- * Always returns new amap with ref_cnt equal to 1.
+ * The size of new amap is determined using `slots`.
+ *
+ * @returns New amap with ref_cnt = 1.
  */
 vm_amap_t *vm_amap_alloc(size_t slots);
 
-/*
+/** Clone the amap.
+ *
  * Create new amap with contents matching old amap. Starting from offset
  * specified by aref and copying specified number of slots.
  *
- * Always returns new amap with ref_cnt equal to 1.
+ * @returns New amap with ref_cnt = 1.
```

Comment:
This change makes all comments consistent in that file. It will be easier to read that file in the future when all the functions descriptions will be made in the same style.